python/lsst/dax/imgserv/api_soda.py

```diff
@@ -124,47 +145,318 @@ def img_capabilities():
+            soda_url = url_for('api_image_soda.img_async', _external=True)
+            return _service_response(soda_url)
+    # new job for request
+    job_id = current_app.soda.do_async(_getparams())
+    return redirect(url_for('api_image_soda.img_async_job',
+                            job_id=job_id,
+                            _external=True))
+
+
+@image_soda.route("/async/<job_id>", methods=["GET"])
+def img_async_job(job_id: str):
+    """ Get the job info, including path to result if ready.
+
+    Parameters
+    ----------
+    job_id: `str`
+
+    Returns
+    -------
+    xml: `str`
+        the job description.
+    """
+    ar = app_celery.AsyncResult(job_id)
+    phase = map_phase_from_state[ar.state]
+    soda_pos, duration, start_time, end_time = "NA", "NA", "NA", "NA"
+    if ar.ready():
+        result = ar.get()
+        start_time = datetime.fromtimestamp(result.get("job_start_time"))
+        end_time = datetime.fromtimestamp(result.get("job_end_time"))
+        duration = (end_time - start_time).total_seconds()
+        soda_pos = result.get("soda_params")["POS"]
+    result_url = url_for('api_image_soda.img_async_job_results_result',
+                         job_id=job_id,
+                         _external=True)
+    resp = make_response(render_template("uws_job_descriptor.xml",
+                                         job_id=job_id,
+                                         job_phase=phase,
+                                         job_start_time=start_time,
+                                         job_end_time=end_time,
+                                         job_duration=duration,
+                                         job_result_id=job_id,
+                                         job_result=result_url,
+                                         soda_pos=soda_pos),
+                         HTTPStatus.OK)
+    resp.headers["Content-Type"] = "text/xml"
+    return resp
+
+
+@image_soda.route("/async/<job_id>/phase", methods=["GET"])
+def img_async_job_phase(job_id: str):
+    """ Get the phase info for the job.
+
+    Parameters
+    ----------
+    job_id : `str`
+
+    Returns
+    -------
+    xml: `str`
+        phase for the job.
+    """
+    ar = app_celery.AsyncResult(job_id)
+    phase = map_phase_from_state[ar.state]
+    return _uws_job_response_plain("PHASE="+phase)
+
+
+@image_soda.route("/async/<job_id>/executionduration", methods=["GET"])
+def img_async_job_duration(job_id: str):
+    """ Get the duration in number of seconds for the job.
+
+    Parameters
+    ----------
+    job_id : `str`
+
+    Returns
+    -------
+    xml: `str`
+        the execution duration parameter value for the job.
+    """
+    ar = app_celery.AsyncResult(job_id)
+    if ar.ready():
+        result = ar.get()
+        execution_duration = result.get("executionduration", "NA")
+        return _uws_job_response_plain("EXECUTION_DURATION=" +
+                                       execution_duration)
```

Comment:
I think this is supposed to be EXECUTIONDURATION, without the underscore here.  Also, since I think this isn't how long it takes to run, but is how long it is allowed to run, we should always have some value for it, even a default, not just when the job is done?

http://www.ivoa.net/documents/UWS/20161024/REC-UWS-1.1-20161024.pdf page 24

Reply:
Fixed.